datasetinsights/io/tracker/mlflow.py

```diff
@@ -80,6 +89,42 @@ def get_mlflow(self):
         logger.info("get mlflow")
         return self.__mlflow
 
+    @staticmethod
+    def _get_variables(client_id=None, host_id=None):
```

Comment:
Why not just split into two methods: `_get_client_id(...)` and `_get_host_id(...)`.  Each method should do one and only one thing. 


Also, it might be easier to use default value pattern like:

```
def __init__(self, clinet_id=None, host_id=None,...):
  client_id = client_id | self._get_client_id(...)
  host_id = host_id | self._get_host_id(...)
  ....
```

This is python's syntax sugar set default value when the variable is None.

Reply:
For end users who do not have mlflow setup, for then both values around | will be None, so will throw below error.
           host_id = host | os.environ.get("MLFLOW_HOST_ID", None)
          TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'

Reply:
In this case host_id is required. Why do you need default value to be None? If host_id is not provided, the initialization method should fail immediately when the constructor is called.

Reply:
- host_id is required only if you have mlflow setup. 
- In case of public users who do not have mlflow setup, for them there host_id will be None in constructor, then it will check in env variable and there also host_id won't be configured so if default value lookup won't be None then it will throw key error so fetching hosted_id using: ' os.environ.get("MLFLOW_HOST_ID", None)'.

Reply:
>host_id is required only if you have mlflow setup

If this is the case, why will `MLFlowTracker` fail to initialize if both client_id and host_id are `None`? 

> In case of public users who do not have mlflow setup for them host_id will be None in constructor, then it will lookup in env variable and there also host_id won't be configured so if default value lookup won't be None then it will throw key error so fetching hosted_id by putting default None: ' os.environ.get("MLFLOW_HOST_ID", None)'.

Let's document this in `MLFlowTracker` class docstring. It should be clear to users what's required.

Reply:
>If this is the case, why will MLFlowTracker fail to initialize if both client_id and host_id are None?
Because mlflow is not setup and hence you don't have both Ids so fail to initialize and trigger Null tracker.